warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,329 @@
+        TreeSet<YearMonthDay> holeDates;
+        MetadataHelper metadataHelper = getMetadataHelper();
+        
+        final QueryData queryData = new QueryData();
+        CloseableIterable<QueryData> results;
+        
+        if (config instanceof ShardQueryConfiguration) {
+            ASTJexlScript queryTree = null;
+            try {
+                queryTree = updateQueryTree(scannerFactory, metadataHelper, dateIndexHelper, (ShardQueryConfiguration) config, query, queryData, settings);
+            } catch (StackOverflowError e) {
+                if (log.isTraceEnabled()) {
+                    log.trace("Stack trace for overflow " + e);
+                }
+                PreConditionFailedQueryException qe = new PreConditionFailedQueryException(DatawaveErrorCode.QUERY_DEPTH_OR_TERM_THRESHOLD_EXCEEDED, e);
+                log.warn(qe);
+                throw new DatawaveFatalQueryException(qe);
+            } catch (NoResultsException e) {
+                if (log.isTraceEnabled()) {
+                    log.trace("Definitively determined that no results exist from the indexes");
+                }
+                
+            }
+            
+            Multimap<String,Type<?>> fieldToDatatypeMap = FetchDataTypesVisitor.fetchDataTypes(metadataHelper,
+                            ((ShardQueryConfiguration) config).getDatatypeFilter(), queryTree, false);
+            
+            try {
+                calculateFieldIndexHoles(metadataHelper, fieldToDatatypeMap, (ShardQueryConfiguration) config);
+            } catch (TableNotFoundException e) {
+                log.error("metadata table was not found " + e.getMessage());
+            }
+            
+            List<FieldIndexHole> fieldIndexHoles = ((ShardQueryConfiguration) config).getFieldIndexHoles();
+            List<ValueIndexHole> valueIndexHoles = ((ShardQueryConfiguration) config).getValueIndexHoles();
+            holeDates = generateStartAndEndDates((ShardQueryConfiguration) config);
+            if ((valueIndexHoles == null && fieldIndexHoles == null) || (valueIndexHoles.size() == 0 && fieldIndexHoles.size() == 0) || holeDates.size() == 0) {
+                results = super.process(config, query, settings, scannerFactory);
+                returnQueryData.addDelegate(results);
+                return returnQueryData;
+            }
+            
+            Boolean firstIteration = true;
+            Date startDate, endDate;
+            
+            for (Iterator<YearMonthDay> it = holeDates.iterator(); it.hasNext();) {
+                if (firstIteration) {
+                    firstIteration = false;
+                    startDate = originalStartDate;
+                    if (it.hasNext()) {
+                        endDate = DateHelper.parse(it.next().getYyyymmdd());
+                    } else
+                        endDate = originalEndDate;
+                } else {
+                    startDate = DateHelper.parse(it.next().getYyyymmdd());
```

Comment:
I think startDate should be simply set to the last endDate here.

Reply:
Thank you.  Its getting late, the vaccine is giving me a headache.  I will take a better look at this tomorrow.  This is a very humbling area of the code for me.  It looks trivial to implement.